include/blmc_robots/clamp.hpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * \file
+ * \copyright Copyright (c) 2020, New York University & Max Planck Gesellschaft.
+ */
+#pragma once
+
+namespace blmc_robots
+{
+/**
+ * @brief Clamp an arbitrary Eigen vector.
+ *
+ * @param vector  The vector that is to be clamped.
+ * @param lower_limit  Lower limit.
+ * @param upper_limit  Upper limit.
+ *
+ * @return  Copy of vector where values below or above the limits are set to the
+ *     corresponding limit values.
+ */
+template <typename Vector>
+Vector clamp(const Vector &vector,
```

Comment:
why is this templated? Because of convenience? I do not mind. I would allow us to ass std::vector clamping if needed I guess.

Reply:
It is templated to allow different lengths of Eigen vectors like `Eigen::Vector2d`, `Eigen::Vector4d`, `Eigen::Matrix<double, 42, 1>` (and probably even 2d-matrices but this is not tested).